docker-compose.yml

```diff
@@ -90,9 +96,20 @@ services:
     links:
       - s3:s3.localhost
     logging: *default-logging
+    volumes: *default-volumes
     ports: []
     stdin_open: true
     tty: true
   redis:
     image: redis
     logging: *default-logging
+    volumes:
+      - redis:/data
+
+volumes:
+  db:
+  redis:
+  bundle:
+  node_modules:
+  packs:
+  rails_cache:
```

Comment:
General docker-compose.yml comments:

- The linked article suggests using `tmpfs` for speed, is that something we should do?
- The article also uses the same `image:` for rails, runner, sidekiq, and webpacker (inherited from `app`/`backend`), where we specify a different name for each. Would we get any benefit from combining them like they do?

Reply:
`tmpfs` was one of the few things from the article that I really didn't understand what it was doing and what the consequences of enabling it was, so I decided to just ignore it for now.

The benefits of using the base image is that there is less duplication.  Right now each container a few of the same components (like `stdin_open` and `tty`).  The downside was that you cannot do `docker-compose up` any longer because there are services that do nothing.  However I took a look at the repo showing off these changes and they actually moved them to `x-...` config options, so I'm gonna take a look at moving to that because then we get the benefits with no downsides.

Reply:
For the second point, I'm talking less about the inheritance of the abstract services and more about the fact that several of their non-abstract services are based on the same [`image`][1].

e.g. theirs is like:
```yaml
services:
  rails:
    image: example-dev:1.0.0
    ...
  runner:
    image: example-dev:1.0.0
    ...
  webpacker:
    image: example-dev:1.0.0
    ...
```
and ours is like
```yaml
services:
  web:
    image: web
    ...
  worker:
    image: worker
    ...
  webpacker:
    image: webpacker
    ...
```
My first reaction was maybe they are able to build once and reuse many times and we are not, but I thought I remembered you showing me once that we do reuse images, so I'm just confused what the difference is between the two setups I guess.


[1]: https://docs.docker.com/compose/compose-file/#image

Reply:
Oh, I believe the benefit to that is that when you switch branches to work on something that might need to change the Dockerfile in some way, you can bump the version and that way anyone testing that branch only needs to build it once versus needing to build it every time they switch to it, and then when they switch back to another (ie: master) having to rebuild off master.

Reply:
I think I understand the advantage of specifying a version number, but what's the advantage of specifying the same string for all the services?

Reply:
Not an advantage per se, but since all of them use the base image you can just tie them all to the same number since you would need to increment them all if you changed the Dockerfile in any way.

Reply:
Pretty sure this question went straight over my head a couple times.  Re-reading it now after working with the production dockerfile I think I understand the question now, but I don't have an answer.  I'm not sure what that name is actually used for in the end.